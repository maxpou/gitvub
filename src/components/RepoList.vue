<template>
  <section class="section--center mdl-grid mdl-grid--no-spacing mdl-shadow--2dp">
    <div class="mdl-card mdl-cell mdl-cell--12-col">
      <div class="mdl-card__supporting-text mdl-grid mdl-grid--no-spacing">
        <h4>Repositories</h4>

        <div
          v-for="(repository, index) in repositories"
          :key="index"
          class="section__text mdl-cell mdl-cell--12-col">

          <router-link
            :to="{ name: 'RepoDetail', params: { user: repository.owner.login, repository: repository.name }}"
            tag="h5"
          >{{ includeOwner ? repository.full_name : repository.name }}</router-link>

          <p>{{ repository.description }}</p>
          <span
            v-if="repository.language"
            class="mdl-chip">
            <span class="mdl-chip__text">{{ repository.language }}</span>
          </span>
          <span
            v-if="repository.license"
            class="mdl-chip mdl-chip--contact">
            <span class="mdl-chip__contact"><i class="material-icons">gavel</i></span>
            <span class="mdl-chip__text">{{ repository.license.spdx_id }}</span>
          </span>
          <span
            v-if="repository.forks_count"
            class="mdl-chip mdl-chip--contact">
            <span class="mdl-chip__contact"><i class="material-icons">call_split</i></span>
            <span class="mdl-chip__text">{{ repository.forks_count }}</span>
          </span>
          <span
            v-if="repository.stargazers_count"
            class="mdl-chip mdl-chip--contact">
            <span class="mdl-chip__contact"><i class="material-icons">star_rate</i></span>
            <span class="mdl-chip__text">{{ repository.stargazers_count }}</span>
          </span>
        </div>
        <div
          v-if="repositories.length === 0"
          class="section__text mdl-cell mdl-cell--12-col">
          <h5>This user doesn't have any repositories.</h5>
        </div>
      </div>
    </div>
  </section>
</template>

<script>
export default {
  name: 'RepoList',
  props: {
    repositories: {
      type: Array,
      default: () => []
    },
    includeOwner: {
      type: Boolean,
      default: false
    }
  }
}
</script>

<style scoped>
h5 {
  cursor: pointer;
  color: #009688;
}

.mdl-chip__contact > .material-icons {
  padding-top: 4px;
  padding-left: 4px;
}

.section__text:not(:last-child) {
  padding-top: 8px;
  padding-bottom: 8px;
}

.section__text:not(:last-child) {
  border-bottom: 1px solid rgba(0,0,0,.13);
}

.section__text > h5 {
  margin-top: 10px;
}

.mdl-card {
  min-height: 220px;
}
</style>
