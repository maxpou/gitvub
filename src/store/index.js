import Vue from 'vue'
import Vuex from 'vuex'
import repositories from './modules/repositories'
import users from './modules/users'
import navigator from './modules/navigator'

Vue.use(Vuex)

const debug = process.env.NODE_ENV !== 'production'

const store = new Vuex.Store({
  modules: {
    repositories,
    users,
    navigator
  },
  strict: debug
})

/* istanbul ignore if */
if (module.hot) {
  module.hot.accept([
    './modules/repositories',
    './modules/users'
  ], () => {
    const repositories = require('./modules/repositories').default
    store.hotUpdate({
      modules: {
        repositories,
        users,
        navigator
      }
    })
  })
}

export default store
