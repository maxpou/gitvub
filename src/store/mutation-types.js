export const RECEIVE_REPOSITORIES = 'RECEIVE_REPOSITORIES'
export const RECEIVE_LANGUAGES = 'RECEIVE_LANGUAGES'
export const FETCH_USER_REPOS = 'FETCH_USER_REPOS'
export const FETCH_REPO_DETAIL = 'FETCH_REPO_DETAIL'
export const RECEIVE_REPOSITORY_LANGUAGE = 'RECEIVE_REPOSITORY_LANGUAGE'
export const RECEIVE_REPOSITORY_README = 'RECEIVE_REPOSITORY_README'
