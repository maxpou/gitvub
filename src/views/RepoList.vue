<template>
  <div class="repoList">

    <profile :user="currentUser" />

    <div class="card-columns">
      <repo-list-item
        v-for="(repository, index) in currentUserRepositories"
        :key="index"
        :repository="repository"
      />
    </div>

    <div>
      <a
        v-if="!isFullyLoaded"
        @click="loadMoreRepositories($route.params.user)"
        class="btn btn-outline-info btn-block">Load more repositories</a>
    </div>

  </div>
</template>

<script>
import { mapGetters, mapActions } from 'vuex'
import Profile from '@/components/Profile'
import RepoListItem from '@/components/RepoListItem'

export default {
  name: 'RepoList',
  components: {
    Profile,
    RepoListItem
  },
  computed: {
    ...mapGetters([
      'currentUserRepositories',
      'currentUser',
      'isFullyLoaded'
    ])
  },
  watch: {
    '$route': 'fetchData'
  },
  created () {
    this.fetchData()
  },
  methods: {
    ...mapActions([
      'loadUserRepositories',
      'loadUser',
      'loadMoreRepositories'
    ]),
    fetchData () {
      this.loadUserRepositories(this.$route.params.user)
      this.loadUser(this.$route.params.user)
    }
  }
}
</script>
