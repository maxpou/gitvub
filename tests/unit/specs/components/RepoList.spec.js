import { mount, RouterLinkStub } from '@vue/test-utils'
import RepoList from '@/components/RepoList.vue'
import store from '@/store'

describe('RepoList.vue', () => {
  it('test initial rendering', () => {
    const wrapper = mount(RepoList, {
      store,
      stubs: {
        'router-link': RouterLinkStub
      },
      propsData: {
        repositories: [{
          name: 'test-repo',
          description: 'I\'m a test repo',
          owner: {
            login: 'bob'
          }
        }]
      }
    })
    const template = wrapper.html()
    expect(template).toMatchSnapshot()
  })

  it('test initial rendering without props', () => {
    const wrapper = mount(RepoList, {
      stubs: {
        'router-link': RouterLinkStub
      }
    })
    const template = wrapper.html()
    expect(template).toMatchSnapshot()
    expect(wrapper.props().repositories).toEqual([])
  })
})
