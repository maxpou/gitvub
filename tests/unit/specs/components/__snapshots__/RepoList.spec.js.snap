// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoList.vue test initial rendering 1`] = `"<section class=\\"section--center mdl-grid mdl-grid--no-spacing mdl-shadow--2dp\\"><div class=\\"mdl-card mdl-cell mdl-cell--12-col\\"><div class=\\"mdl-card__supporting-text mdl-grid mdl-grid--no-spacing\\"><div class=\\"section__text mdl-cell mdl-cell--12-col\\"><h5>test-repo</h5> <p>I'm a test repo</p> <!----> <!----> <!----> <!----></div> <!----></div></div></section>"`;

exports[`RepoList.vue test initial rendering without props 1`] = `"<section class=\\"section--center mdl-grid mdl-grid--no-spacing mdl-shadow--2dp\\"><div class=\\"mdl-card mdl-cell mdl-cell--12-col\\"><div class=\\"mdl-card__supporting-text mdl-grid mdl-grid--no-spacing\\"> <div class=\\"section__text mdl-cell mdl-cell--12-col\\"><h5>This user doesn't have any repositories.</h5></div></div></div></section>"`;
